compiler/ast.nim

```diff
@@ -1356,6 +1356,17 @@ proc skipTypesOrNil*(t: PType, kinds: TTypeKinds): PType =
     if result.len == 0: return nil
     result = lastSon(result)
 
+proc repeatsInNode*(symbol: PSym; n: PNode): bool =
```

Comment:
In ``types.nim`` we have an iterator over types that can detect illegal recursions.